### ISSUE TITLE
[Bug]: Body's specific headers overrides all the request/client set headers

### ISSUE THREAD

**Issue by dariopellegrini** [immobiliare/RealHTTP#62]
### Platform Version

iOS 16

### SDK Version

1.7.2

### Xcode Version

Xcode 14

### Steps To Reproduce

I have this request

```swift
try await HTTPRequest {
            $0.path = "businesscards"
            $0.headers = HTTPHeaders([
                .authorization:"Bearer \(token)",
                .contentType:"multipart/form-data"
            ])
            
            $0.method = .post
            $0.body = try .multipart(boundary: nil, { form in
                if let name = businessCardCreation.name {
                    try form.add(string: name, name: "name")
                }
                if let surname = businessCardCreation.surname {
                    try form.add(string: surname, name: "surname")
                }
                if let email = businessCardCreation.email {
                    try form.add(string: email, name: "email")
                }
            })
        }.fetch().decode(BusinessCard.self, decoder: Decoders.decoder)
```

that is not sending Authorization header to the backend.

### Expected Behavior

I expect to see the Authorization header in the backend.

### Actual Incorrect Behavior

I can't see the authorization header in the backend.
Here is a screenshot of the headers sent in a local call, but the same happens on the cloud.

<img width="732" alt="Schermata 2022-10-07 alle 07 36 56" src="https://user-images.githubusercontent.com/6443619/194475306-261f196d-eb40-4657-9caa-bccea6d2f244.png">

Trying the same request using Alamofire everything works.
<img width="747" alt="Schermata 2022-10-07 alle 08 02 21" src="https://user-images.githubusercontent.com/6443619/194478551-7bbf359e-93ca-4235-a8eb-e70418f1a645.png">

**Comment by nashfive** [comment]
Have you tried to use the `HTTPHeader.authBearerToken("")` instead ?

**Comment by malcommac** [comment]
Hi @dariopellegrini , thank you for your report.
It's a bug: due to a design choice, the `headers` property is just a forward to `body.headers` property, so when you assign the `body` to the request, it will override any previously set `headers` property.
We'll provide a fix ASAP. Meanwhile, a possible workaround could be to set the `$0.headers=` after the body, and it should work fine.

**Comment by malcommac** [comment]
It will be merged in #63 with a minor release marked as 1.7.3.  
An additional unit test was also added to mark and validate any further changes.
Thank you very much for your report.

**Comment by malcommac** [comment]
1.7.3 was published few minutes ago both and it's available both on SPM and CocoaPods.

**Comment by dariopellegrini** [comment]
@malcommac
I confirm that the bug is fixed in 1.7.3.

Thank you!
